upgrade.go

```diff
@@ -250,26 +275,16 @@ func (b *versionedKVBackend) Upgrade(ctx context.Context, s logical.Storage) err
 		}
 		b.l.Unlock()
 
-		// Write upgrade done value
-		upgradeInfo.Done = true
-		info, err := proto.Marshal(upgradeInfo)
-		if err != nil {
-			b.Logger().Error("encoding upgrade info resulted in an error", "error", err)
-		}
-
-		err = s.Put(ctx, &logical.StorageEntry{
-			Key:   path.Join(b.storagePrefix, "upgrading"),
-			Value: info,
-		})
-		if err != nil {
-			b.Logger().Error("writing upgrade done resulted in an error", "error", err)
-		}
-
+		writeUpgradeInfoDoneFunc()
```

Comment:
Since we're calling this here do we still need the loop at the top of this func?

Reply:
We do, yeah - the loop at the top is for the 'currently upgrading' entry, and this function adds the 'we're done upgrading' entry. We only write the second in the sync case, but writing both for a long upgrade is still relevant.

Reply:
Ah write, I missed the difference in `Done` value. Whoops! Thanks for the clarification.